File: SDKs/twilio.com/twilio_video_v1/1.36.0/typescript/src/sdk/models/operations/listroom.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTROOM_SERVERS = [
	"https://video.twilio.com",
];



export class ListRoomQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=DateCreatedAfter" })
  dateCreatedAfter?: Date;

  @Metadata({ data: "queryParam, style=form;explode=true;name=DateCreatedBefore" })
  dateCreatedBefore?: Date;

  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=Status" })
  status?: shared.RoomEnumRoomStatusEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=UniqueName" })
  uniqueName?: string;
}


export class ListRoomSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListRoomRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  queryParams: ListRoomQueryParams;

  @Metadata()
  security: ListRoomSecurity;
}


export class ListRoomListRoomResponseMeta extends SpeakeasyBase {
  @Metadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @Metadata({ data: "json, name=key" })
  key?: string;

  @Metadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class ListRoomListRoomResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=meta" })
  meta?: ListRoomListRoomResponseMeta;

  @Metadata({ data: "json, name=rooms", elemType: shared.VideoV1Room })
  rooms?: shared.VideoV1Room[];
}


export class ListRoomResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listRoomResponse?: ListRoomListRoomResponse;

  @Metadata()
  statusCode: number;
}

File: SDKs/twilio.com/twilio_video_v1/1.37.3/go/pkg/models/operations/listroom.go
package operations

import (
	"openapi/pkg/models/shared"
	"time"
)

var ListRoomServerList = []string{
	"https://video.twilio.com",
}

type ListRoomQueryParams struct {
	DateCreatedAfter  *time.Time                     `queryParam:"style=form,explode=true,name=DateCreatedAfter"`
	DateCreatedBefore *time.Time                     `queryParam:"style=form,explode=true,name=DateCreatedBefore"`
	PageSize          *int64                         `queryParam:"style=form,explode=true,name=PageSize"`
	Status            *shared.RoomEnumRoomStatusEnum `queryParam:"style=form,explode=true,name=Status"`
	UniqueName        *string                        `queryParam:"style=form,explode=true,name=UniqueName"`
}

type ListRoomSecurity struct {
	AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
}

type ListRoomListRoomResponseMeta struct {
	FirstPageURL    *string `json:"first_page_url,omitempty"`
	Key             *string `json:"key,omitempty"`
	NextPageURL     *string `json:"next_page_url,omitempty"`
	Page            *int64  `json:"page,omitempty"`
	PageSize        *int64  `json:"page_size,omitempty"`
	PreviousPageURL *string `json:"previous_page_url,omitempty"`
	URL             *string `json:"url,omitempty"`
}

type ListRoomListRoomResponse struct {
	Meta  *ListRoomListRoomResponseMeta `json:"meta,omitempty"`
	Rooms []shared.VideoV1Room          `json:"rooms,omitempty"`
}

type ListRoomRequest struct {
	ServerURL   *string
	QueryParams ListRoomQueryParams
	Security    ListRoomSecurity
}

type ListRoomResponse struct {
	ContentType      string
	ListRoomResponse *ListRoomListRoomResponse
	StatusCode       int64
}
